airflow/providers/apache/spark/decorators/pyspark_submit.py

```diff
@@ -0,0 +1,205 @@
+                    raise AirflowException(
+                        f"Invalid key '{key}' in op_kwargs. You don't need to set it because it's a "
+                        "variable that will be automatically set within the Python process of the Spark "
+                        "job submitted via spark-submit."
+                    )
+                warnings.warn(
+                    f"Invalid key '{key}' in op_kwargs. You don't need to set it because it's a "
+                    "variable that will be automatically set within the Python process of the Spark "
+                    "job submitted via spark-submit.",
+                    UserWarning,
+                    stacklevel=2,
+                )
+
+        super().__init__(
+            python_callable=python_callable,
+            op_args=op_args,
+            op_kwargs=op_kwargs,
+            **kwargs,
+        )
+
+    def execute(self, context: Context):
+        with TemporaryDirectory() as tmp_dir:
+            script_filename = os.path.join(tmp_dir, "script.py")
+            input_filename = os.path.join(tmp_dir, "SCRIPT__GENERATED__AIRFLOW.IN")
+
+            if self.op_args or self.op_kwargs:
+                with open(input_filename, "wb") as file:
+                    self.pickling_library.dump({"args": self.op_args, "kwargs": self.op_kwargs}, file)
+                files = (self.files or "").split(",")
+                self.files = ",".join(files + [input_filename])
+
+            parameters = inspect.signature(self.python_callable).parameters
+            use_spark_context = use_spark_session = False
+            if "sc" in parameters:
+                use_spark_context = True
+            if "spark" in parameters:
+                use_spark_session = True
+
+            py_source = dedent(
+                f"""\
+                from pyspark import SparkFiles
+                from pyspark.sql import SparkSession
+
+                # Script
+                {{python_callable_source}}
+
+                if {bool(self.op_args or self.op_kwargs)}:
+                    with open(SparkFiles.get("{os.path.basename(input_filename)}"), 'rb') as file:
+                        arg_dict = {self.pickling_library.__name__}.load(file)
+                else:
+                    arg_dict = {{default_arg_dict}}
+
+                if {use_spark_session}:
+                    arg_dict["kwargs"]["spark"] = SparkSession.builder.getOrCreate()
+                if {use_spark_context}:
+                    spark = arg_dict.get("spark") or SparkSession.builder.getOrCreate()
+                    arg_dict["kwargs"]["sc"] = spark.sparkContext
+
+                # Call
+                {self.python_callable.__name__}(*arg_dict["args"], **arg_dict["kwargs"])
+
+                # Exit
+                exit(0)
+                """
+            ).format(
+                python_callable_source=self.get_python_source(), default_arg_dict='{"args": [], "kwargs": {}}'
+            )
```

Comment:
To enable reading input file from spark job submitted with spark-submit, it needs to be passed using `--files` option of spark-submit. 
However, the actual path of files passed via `--files` option is determined at runtime (under /tmp somewhere).

In current `write_python_script` function, which uses [Jinja2 template file](https://github.com/apache/airflow/blob/main/airflow/utils/python_virtualenv_script.jinja2#L54-L73), `sys.argv[2]` is hardcoded. This makes it seemingly impossible to pass op_args and op_kwargs as Jinja2 context.

Therefore, I crafted the execution statement directly and forced termination with `exit(0)`.

https://github.com/apache/airflow/blob/e827bfbd2c79f3432411e14351c245a06410766a/airflow/utils/python_virtualenv_script.jinja2#L54-L73